feat(js): Add `skipBrowserExtensionCheck` documentation (#11693)
getsentry/sentry-docs: docs/platforms/javascript/common/best-practices/shared-environments.mdx

- Libraries
- Any other scenario where you might have multiple Sentry instances running in the same environment

<Alert level="danger">

When setting up Sentry in a shared environment where multiple Sentry instances may run, you should **not use `Sentry.init()`**, as this will pollute the global state. If your browser extension uses `Sentry.init()`, and a website also uses Sentry, the extension may send events to the website's Sentry project, or vice versa.

</ Alert>

## Shared Environment Setup

For the use cases listed above, set up a client manually as seen in the example below.
In addition, avoid adding any integrations that use global state, like `Breadcrumbs` or `GlobalHandlers`.
Furthermore, some default integrations that use the global state have to be filtered as in the example below.

> As a rule of thumb, it's best to avoid using any integrations and to rely on manually capturing errors only in such scenarios.

Reviewer: ```suggestion
In these scenarios, it's a best practice to avoid using any integrations and to rely on manually capturing errors.
```
This is a very opinionated scenario and feel free to reject it, but I got a little lost trying to parse this sentence.

// filter integrations that use the global variable
const integrations = getDefaultIntegrations({}).filter((defaultIntegration) => {
  return !["BrowserApiErrors", "Breadcrumbs", "GlobalHandlers"].includes(

>     defaultIntegration.name

Author: auto format kicked in here

You can disable that by going to your **Project Settings > Inbound Filters** and disabling filtering out errors known to be caused by browser extensions.

Read more about Inbound Filters in the product documentation under [Inbound filters](/concepts/data-management/filtering/#inbound-data-filters).

## Skipping the Browser Extension Check

_Available in all browser-based SDKs since version `8.37.0`_

If for some reason, the SDK wrongfully detects that it's initialized in a browser extension, you can skip the check by specifying the `skipBrowserExtensionCheck` option when initializing the SDK:

```javascript
import * as Sentry from "@sentry/browser";

Sentry.init({
  dsn: "___PUBLIC_DSN___",
  skipBrowserExtensionCheck: true,
  // ...
});
```

<Alert level="danger">

> Don't use this option light-hearted if you're in fact using the SDK in a browser extension or another shared environment.

Reviewer: ```suggestion
You shouldn't use this option if you're in fact using the SDK in a browser extension or another shared environment.
```

Initializing the SDK via `Sentry.init` has no advantages over manually setting up the client and scope as described on this page.
You'd risk quota increases with unactionable issues, interference with other Sentry SDKs and data leakage by doing so.

</Alert>

This option is purely meant as an escape hatch if our browser extension check is incorrectly detecting a browser extension.

> An example for this might be a

Reviewer 2: I think there is something missing

Author: lol good catch, thx :) Finished the sentence now

> An example for this might be a cross-platform application framework that exposes similar global APIs like browser a browser extension environment.

Author: ```suggestion
An example for this might be a cross-platform application framework that exposes similar global APIs like browser extensions.
```

</Alert>

This option is purely meant as an escape hatch if our browser extension check is incorrectly detecting a browser extension.

> An example for this might be a cross-platform application framework that exposes similar global APIs like browser a browser extension environment.

Reviewer 2: Something is off here or am I missing something? 🤔 
"like browser a browser extension"
